File: simulators/de.unistuttgart.iste.sqa.mpw.karasimulator.java/examples/src/main/java/de/unistuttgart/kara/examples/Example02.java
package de.unistuttgart.kara.examples;

import de.unistuttgart.kara.main.SimpleKaraGame;

public class Example02 extends SimpleKaraGame {
    public static void main(final String[] args) {
        createInstance(Example02.class);
    }

    @Override
    protected String getWorldFile() {
        return "/worlds/example02.ter";
    }

    /**
     * Another kara program.
     */
    @Override
    protected void run() {
        displayInNewGameWindow();

        kara.turnRight();

        for (int i = 0; i < 3; i++) {
            kara.move();
            kara.putLeaf();
        }

        rotate180Degrees();

        for (int i = 0; i < 3; i++) {
            kara.removeLeaf();
            kara.move();
        }
    }

    private void rotate180Degrees() {
        kara.turnLeft();
        kara.turnLeft();
    }
}

File: simulators/de.unistuttgart.iste.sqa.mpw.karasimulator.cpp/examples/Example02.cpp
#include "SimpleKaraGame.h"

using namespace kara;

class Example02 : public karasimulator::SimpleKaraGame {
protected:
    void run() override;

    void rotate180Degrees();
};

void Example02::run() {
    loadWorldFromResourceFile("resources/worlds/example02.ter");
    startGame();
    
    kara.turnRight();

    for (int i = 0; i < 3; i++) {
        kara.move();
        kara.putLeaf();
    }

    rotate180Degrees();

    for (int i = 0; i < 3; i++) {
        kara.removeLeaf();
        kara.move();
    }
}

void Example02::rotate180Degrees() {
    kara.turnLeft();
    kara.turnLeft();
}

int main() {
    karasimulator::SimpleKaraGame::createInstance<Example02>();
    return 0;
}
